src/safeposix/vmmap.rs

```diff
@@ -0,0 +1,626 @@
+use super::vmmap_constants::*;
+use std::io;
+use nodit::NoditMap;
+use nodit::{interval::ie, Interval};
+
+/// Used to identify whether the vmmap entry is backed anonymously,
+/// by an fd, or by a shared memory segment
+
+#[allow(dead_code)]
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum MemoryBackingType {
+    None, // just a dummy value for places where it needs to be passed, but you dont have the value
+    Anonymous,
+    SharedMemory(u64),   // stores shmid
+    FileDescriptor(u64), // stores file descriptor addr
+}
+
+
+/// An entry in the virtual memory map that contains fields such as page number, number of pages, 
+/// permissions, file offset, file size, shared memory ID, and backing fields to distinguish memory types.
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct VmmapEntry {
+    pub page_num: u32, /* base virtual addr >> NACL_PAGESHIFT */
+    pub npages: u32,   /* number of pages */
+    pub prot: i32,     /* mprotect attribute */
+    pub maxprot: i32,
+    pub flags: i32,       /* mapping flags */
+    pub removed: bool,    /* flag set in fn Update(); */
+    pub file_offset: i64, /* offset into desc */
+    pub file_size: i64,   /* backing store size */
+    pub cage_id: u64,
+    pub backing: MemoryBackingType,
+}
+
+
+// Implement methods for VmmapEntry
+// Constructor to create a new VmmapEntry
+#[allow(dead_code)]
+impl VmmapEntry {
+    pub fn new(
+        page_num: u32,
+        npages: u32,
+        prot: i32,
+        maxprot: i32,
```

Comment:
I think we can just set this on create since it shouldnt be able to be modified after